Rate unit test

mike1796 opened the pull request.

mike1796 committed:
Adding of unit test

mike1796 committed:
Adding of unit test

mike1796 committed:
Unit + behat ok

quentinus95 commented on .travis.yml:
```diff
@@ -21,5 +21,5 @@ before_script:
 script:
     - composer install --no-interaction --no-progress
-    - ./vendor/bin/behat
+    - ./bin/test.sh
```
can you remove the `.sh` extension to keep the bin folder consistent?

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+    private $accessListener;
+
+    /**
+     * {@inheritdoc}
```
this annotation block can be removed

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+            $this->accessListener->onKernelRequest($event);
+        }
+
+        $this->assertEquals($event->getResponse()->getStatusCode(), 429);
```
If I'm not wrong, there is a constant in the Request class (from Http Foundation) to be used instead of the raw integer value.

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+
+        $this->assertEquals($event->getResponse()->getStatusCode(), 429);
+
+        $ipRequest = $this->em->getRepository('AppBundle:IpRequest')->findOneBy(['ip' => '127.0.0.1']);
```
this should be performed before the previous assertion to keep the test structured (initialization / assertion)

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+        $event = $this->getEvent();
+        $this->accessListener->onKernelRequest($event);
+        $ipRequest = $this->em->getRepository('AppBundle:IpRequest')->findOneBy(['ip' => '127.0.0.1']);
+        $this->assertNull($event->getResponse());
```
one line break before this line please :)

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    /**
+     * {@inheritdoc}
```
you can remove this annotation block

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+        parent::tearDown();
+
+        $this->em->close();
+        $this->em = null; // avoid memory leaks
```
useless stackoverflow(?) copy pasted comment

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+
+    public function testRequestPassedAgain()
+    {
+        sleep(AccessListener::AGGREGATION_DELAY);
```
I guess, if we sleep here, that you expect this test to be run after the previous method of this class? If so, you should use the @depends annotation (see PHPUnit documentation)

mike1796 committed:
Correction of differents points for review

 - Remove sh extension in bin directory
 - Remove annotation block in test
 - fix test structure (initialization/assertion)
 - Depends on tests
 - Remove copy pasted comment

mike1796 committed:
Merge branch 'master' into rate_unit_test

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+            ->get('doctrine')
+            ->getManager();
+
+        $this->accessListener = new AccessListener($this->em, 'unit-test');
```
you can get the current environment from static::$kernel. It should be `test`.

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+     */
+    private $em;
+
+    private $accessListener;
```
A cleaner way would be to remove the private attribute, create the instance in the first test and return it to the next dependant test.

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+    public function testRequestPassed()
+    {
+        $event = $this->getEvent();
+        $this->accessListener->onKernelRequest($event);
```
Create an access listener here and return it at the end of the method.

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+    /**
+     * @depends testRequestPassed
+     */
+    public function testRequestIsBlocked()
```
Get the AccessListener as method argument.

quentinus95 commented on tests/AppBundle/Listener/AccessListenerTest.php:
```diff
@@ -0,0 +1,106 @@
+    /**
+     * @depends testRequestIsBlocked
+     */
+    public function testRequestPassedAgain()
```
Get the AccessListener as an arguement.

quentinus95 commented:
@mike1796 just some minor changes to perform and it should be ready for merging.

mike1796 committed:
Clean up before merge

AccessListener is not a member variable anymore

mike1796 committed:
Merge branch 'master' into rate_unit_test

mike1796 committed:
Fix of lint for travis build

quentinus95 commented:
Good!

quentinus95 commented:
But tests are failing :cry:

mike1796 committed:
Test are fixed

Adding phpunit to composer
Don't close em in tearDown function
